parsl/tests/test_regression/test_854.py

```diff
@@ -0,0 +1,67 @@
+import time
+import multiprocessing
+import pytest
+from parsl.executors.high_throughput.mac_safe_queue import MacSafeQueue
+import random
+
+
+def consumer(in_q, out_q, delay=0):
+    import time
+    while True:
+        x = in_q.get()
+        time.sleep(delay)
+        if x == 'STOP':
+            out_q.put('STOPPED')
+            break
+        else:
+            out_q.put(x)
+
+
+@pytest.mark.local
+def test_mac_safe_queue():
+    """ Regression test for HTEX being broken on Mac OS: https://github.com/Parsl/parsl/issues/854
+    This test doesn't test the fix on mac's however it tests a multiprocessing queue replacement
+    that is safe to run on Mac OS.
+    """
+    task_q = MacSafeQueue()
+    result_q = MacSafeQueue()
+
+    p = multiprocessing.Process(target=consumer, args=(task_q, result_q,))
+    p.start()
+    for i in range(10):
+        task_q.put(i)
+        result_q.get()
+    task_q.put('STOP')
+    r = result_q.get()
+    assert r == 'STOPPED', "Did not get stopped confirmation, got:{}".format(r)
+    p.terminate()
+
+
+@pytest.mark.local
+def test_mac_safe_queue_size():
+    """ Regression test for HTEX being broken on Mac OS: https://github.com/Parsl/parsl/issues/854
+    This test doesn't test the fix on mac's however it tests a multiprocessing queue replacement
+    that is safe to run on Mac OS.
+    """
+    task_q = MacSafeQueue()
+    result_q = MacSafeQueue()
+
+    x = random.randint(0, 100)
+
+    [task_q.put(i) for i in range(x)]
+    assert task_q.empty() is False, "Task queue should not be empty"
+    assert task_q.qsize() == x, "Task queue should not be empty"
+
+    p = multiprocessing.Process(target=consumer, args=(task_q, result_q,))
+    p.start()
+    task_q.put('STOP')
+    time.sleep(0.1)
```

Comment:
Can this be a join on p rather than a time?

Reply:
Fixed.